python-backend/app/api/now_applications/resources/now_application_resource.py

```diff
@@ -29,17 +30,33 @@ class NOWApplicationResource(Resource, UserMixin):
     def get(self, application_guid):
         original = request.args.get('original', False, type=bool)
 
-        now_application_identity = NOWApplicationIdentity.query.filter_by(
-            now_application_guid=application_guid).first()
+        now_application_identity = NOWApplicationIdentity.find_by_guid(application_guid)
         if not now_application_identity:
             raise NotFound('No identity record for this application guid.')
 
         if now_application_identity.now_application_id and not original:
-            application = NOWApplication.find_by_application_id(
-                now_application_identity.now_application_id)
+            application = now_application_identity.now_application
             application.imported_to_core = True
         else:
             application = transmogrify_now(now_application_identity)
             application.imported_to_core = False
 
-        return application
\ No newline at end of file
+        return application
+
+    @api.doc(
+        description=
+        'Updates a now application and nested objects, this endpoint is not idempotent, nested objects without primary keys will be treated as new objects.'
+    )
+    @requires_role_edit_permit
+    @api.marshal_with(NOW_APPLICATION_MODEL, code=200)
+    def put(self, application_guid):
+        now_application_identity = NOWApplicationIdentity.find_by_guid(application_guid)
+        if not now_application_identity:
+            raise NotFound('No identity record for this application guid.')
+
+        if now_application_identity.now_application_id is None:
+            raise NotImplemented('this would cause transmogrify and save, not yet')
+        data = request.json
+        now_application_identity.now_application.deep_update_from_dict(data)
```

Comment:
Should this just be named update? Does this class need to know implementation details of the other class?

Reply:
i think 'deep' is an important prefix, because it can be unclear if it's updating the object, or related objects as well. could remove the 'from' dict suffix, but without that the caller doesn't know what the input type should be.